File: code/my_fschool_frontend/lib/model/response/application_response.dart
import 'package:dart_mappable/dart_mappable.dart';

part 'application_response.mapper.dart';

@MappableClass(
  caseStyle: CaseStyle.camelCase,
  generateMethods:
      GenerateMethods.decode, // Chỉ tạo hàm decode (fromJson/fromMap) giống mẫu
)
class ApplicationResponse with ApplicationResponseMappable {
  final String id;
  final String
  title; // Đã được BE dịch sang Tiếng Việt (e.g., 'Đơn xin nghỉ học')
  final String status; // 'PENDING', 'APPROVED', 'REJECTED'
  final String sentDate; // Định dạng chuỗi 'dd/MM/yyyy HH:mm'
  final String? processedDate; // Định dạng chuỗi 'dd/MM/yyyy HH:mm' (nullable)
  final String? handlerName; // Họ tên thầy cô duyệt đơn (nullable)
  final String? fromDate; // Ngày bắt đầu nghỉ 'dd/MM/yyyy' (nullable)
  final String? toDate; // Ngày kết thúc nghỉ 'dd/MM/yyyy' (nullable)
  final String requestContent; // Lý do phụ huynh viết đơn
  final String? responseContent; // Phản hồi từ nhà trường (nullable)

  const ApplicationResponse({
    required this.id,
    required this.title,
    required this.status,
    required this.sentDate,
    this.processedDate,
    this.handlerName,
    this.fromDate,
    this.toDate,
    required this.requestContent,
    this.responseContent,
  });

  static ApplicationResponse fromJson(String json) =>
      ApplicationResponseMapper.fromJson(json);

  static ApplicationResponse fromMap(Map<String, dynamic> map) =>
      ApplicationResponseMapper.fromMap(map);
}

File: code/my_fschool_backend/src/main/java/prm393/se1911/assignment/myfschoolbackend/model/response/ApplicationResponse.java
package prm393.se1911.assignment.myfschoolbackend.model.response;

import java.util.UUID;

public record ApplicationResponse(
        UUID id,
        String title,            // Bản chất là applicationType chuyển sang tiếng Việt hoặc giữ nguyên tùy phen
        String status,           // 'PENDING', 'APPROVED', 'REJECTED'
        String sentDate,         // submittedAt định dạng 'dd/MM/yyyy HH:mm'
        String processedDate,    // processedAt định dạng 'dd/MM/yyyy HH:mm' (nếu có)
        String handlerName,      // Họ tên Giáo viên xử lý lấy từ bảng User qua handler_id
        String fromDate,         // Định dạng 'dd/MM/yyyy' (nếu có)
        String toDate,           // Định dạng 'dd/MM/yyyy' (nếu có)
        String requestContent,   // Lý do (reason) gửi đơn
        String responseContent   // Phản hồi từ nhà trường (schoolResponse)
) {
}
